src/backend/types.js

```diff
@@ -90,6 +90,17 @@ export type ProfilingSummary = {|
   rootID: number,
 |};
 
+type ExportedProfilingDataFromRendererV1 = {|
```

Comment:
I think `ExportedProfilingData` as a type with a `version: number` field makes more sense.

I don't _think_ we'll necessarily want to support multiple versions. If we add new fields or change the data in some meaningful way, that will break old exports.

Reply:
My idea here is to express the intent with data type names very clearly. Even if you won't support multiple versions in the same codebase, you may still want to know what is type-compatible with what, so that the compiler tells you what you might have broken. Aa you might have noticed, this type is not exported (on purpose; I should've left a comment about my design decision here).

Reply:
Not sure I see the value of that type, since there's no way to enforce it. (We would still need a runtime check of the data version to make sure it's the one the current build supports. This data comes from an external source; it isn't defined in a Flow parsed file)

Reply:
> Not sure I see the value of that type, since there's no way to enforce it.

The value is that you describe what is expected there. I'm coming from TypeScript background where I had `io-ts` for runtime type enforcement and derived compile-time types. I think it has Flow, too, if you want to look into making I/O boundaries typed.

> (We would still need a runtime check of the data version to make sure it's the one the current build supports. This data comes from an external source; it isn't defined in a Flow parsed file)

You're right. In my vision there's a clear boundary (I call it I/O boundary) between the untyped world (JSON, files) and typed world (Flow), and there should definitely be functions that ensure the types and pipe data from untyped world to typed, serve as a single entry point into the typed world.

Reply:
Circling back. My concern is that I don't see the value in naming a type e.g. v1 if we only intend to support one type at any given time. I think the type name maybe muddies the water since (to me) it implies that we intend to support more than one version.

Reply:
Renamed the types to not contain `V1`.